__tests__/user/user.query.test.js

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-undef */
+const { gql } = require('apollo-boost');
+const client = require('../../utils/apollo-client');
+
+require('dotenv').config();
+
+let token, userId
+
+describe('queries', () => {
+    beforeAll(async () => {
+        const register = await client.mutate({
+            mutation: gql`
+                mutation {       
+                    registerUser(user: {
+                        firstName: "Test" 
+                        lastName: "User"
+                        email: "test.email@gmail.com"
+                        password: "12345678Te"
+                    }) {
+                        _id
+                        firstName
+                        lastName
+                        email
+                        role
+                        registrationDate
+                        credentials {
+                            tokenPass
+                        } 
+                    }
+                }
+            `
+        })
+
+        userId = register.data.registerUser._id
+
+        const authRes  = await client.mutate({
+            mutation: gql`
+                mutation {       
+                    loginUser(user: {
+                        email: "test.email@gmail.com"
+                        password: "12345678Te"
+                    }) {
+                        token
+                    }
+                }
+            `
+        })
+        token = authRes.data.loginUser.token
+    });
+
+    afterAll(async () => {
+        await client.mutate({
+            mutation: gql`
+                mutation($userId: ID!) {       
+                    deleteUser(id: $userId) {
+                        _id
+                    }
+                }
+            `,
+            variables: {
+                userId
+            }
+        })
+    });
+
+    test('should recive all users', async () => {
+        const res = await client.query({
+            query: gql`
+                query {       
+                    getAllUsers{
+                        email
+                        firstName
+                        lastName
+                        phoneNumber
+                    }
+                }
+            `
+        })
+         expect(res.data.getAllUsers).toContainEqual({
+            "firstName": "Test" ,
+            "lastName": "User",
+            "email": "test.email@gmail.com",
+            "phoneNumber": null        
+        });
+
+    })
+
+    test('should recive user by token', async () => {
```

Comment:
Please, add a test that checks that proper errors are returned if client provides the invalid authorization token

Reply:
fixed